unittests/test_utility.py

```diff
@@ -1567,6 +1567,9 @@ def __init__(self, x, y):
         self.z = None
         self.w = {1, 2}
 
+        # Dump dict with tuples as keys
+        self.v = {(1, 2): 1}
```

Comment:
This has to be also included in the equality test.

Reply:
In which case it would call for your other point above about `decode()`.